system-addon/content-src/components/Search/Search.jsx

```diff
@@ -8,33 +8,43 @@ const {actionCreators: ac} = require("common/Actions.jsm");
 class Search extends React.Component {
   constructor(props) {
     super(props);
+    this.controller = null;
     this.onClick = this.onClick.bind(this);
-    this.onInputMount = this.onInputMount.bind(this);
+    this.onFocus = this.onFocus.bind(this);
   }
-
   handleEvent(event) {
     // Also track search events with our own telemetry
     if (event.detail.type === "Search") {
       this.props.dispatch(ac.UserEvent({event: "SEARCH"}));
     }
   }
+  initializeSearch() {
+    if (this.controller) {
+      return;
+    }
+    this.controller = new ContentSearchUIController(this.input, this.input.parentNode,
+      "newtab", "newtab");
+  }
   onClick(event) {
+    if (!this.controller) {
+      this.initializeSearch();
+    }
```

Comment:
I'm suspicious if this works as async messages take time to happen ? Although maybe the controller is smart enough to delay its `search`? Also, I suppose it's safe to check if `controller` exists but `initializeSearch` does it too. ;) Whatever you think is clearer.